docs/TESTING.md

```diff
@@ -1,53 +1,60 @@
 
-(As well as the below, there is a rudimentary "single, automatic, end-to-end coinjoin test" in the file test/test_full_coinjoin.py. Recommended to run it
-with -s to see log output; what's "rudimentary" here is that errors may simply result in the process hanging, so you'll want to investigate in that case.
-Use command line:
+Then copy the `regtest_joinmarket.cfg` file from the `test/` directory to the root directory, you probably won't need to change anything here except perhaps the above password, and the `native` setting if you're doing bech32 wallet tests.
 
-    pytest --btcconf=/path/to/bitcoin.conf --btcroot=/path/to/bitcoin/bin/ --btcpwd=whatever --nirc=2 test/test_full_coinjoin.py -s
+Run the test suite via pytest:
 
-)
+    (jmvenv)$ pytest --btcconf=/path/to/bitcoin.conf --btcroot=/path/to/bitcoin/bin/ --btcpwd=whatever --nirc=2 --ignore test/test_full_coinjoin.py -p no:warnings
+
+#### Running tests of sendpayment and tumbler (including with malicious makers)
 
 The file `test/ygrunner.py` provides the ability to spin up a set of yieldgenerator
 bots against the local IRC instance with the local regtest blockchain. It can be
 started with
 
     pytest --btcroot=/path/to/bitcoin/bin/ --btcpwd=123456abcdef --nirc=2 test/ygrunner.py -s
 
-Here the `-s` flag is useful because it prints log output to the console. If you
+Here the `-s` flag is necessary because it prints log output to the console. If you
 keep the logging level at the default `INFO` only a minimum amount will come out, if
 you want more then enter this into the `joinmarket.cfg` in the root directory:
 
     [LOGGING]
     console_log_level = DEBUG
 
-It will print out a hex seed for a wallet you can use for tumble/sendpayment. Next,
-go into the `scripts/` directory and make sure you have copied the `regtest_joinmarket.cfg`
+It will print out a hex seed for a wallet you can use for tumble/sendpayment (just set the walletname to that hex value).
+Next, go into the `scripts/` directory and make sure you have copied the `regtest_joinmarket.cfg`
 file into that directory also, make any changes needed (like the LOGGING one above),
-and run either sendpayment or tumbler with whatever parameters you choose.
+and run either sendpayment or tumbler with whatever parameters you choose, BUT: remember to add the `--datadir=.` argument so that your test `joinmarket.cfg` file gets picked up, not the one in `~/.joinmarket`.
+
+So for example:
+
+```
+(jmvenv) $python sendpayment.py -N2 -m1 5c88acf2546bd7b083b9cfb2e0af7f2d --datadir=. 30000000 2Address
+```
 
 To change the parameters of the yieldgenerators you can edit the parametrization of
-the function `test_start_ygs` in [this file](https://github.com/AdamISZ/joinmarket-clientserver/blob/master/test/ygrunner.py).
+the function `test_start_ygs` in [this file](https://github.com/Joinmarket-Org/joinmarket-clientserver/blob/master/test/ygrunner.py).
```

Comment:
I am sorry for the flooding here.

Because it is not very intuitive, maybe add something like:
`where 5c88acf2546bd7b083b9cfb2e0af7f2d is a 32 character hex value, such as the one given to you when you launched ygrunner.py`.

And replace 2Address with `bcrt1qk48xnvsd9yma2yqgdxhfpcgny7fsz303z0hfv9`?

Reply:
> Because it is not very intuitive, maybe add something like:
where 5c88acf2546bd7b083b9cfb2e0af7f2d is a 32 character hex value used as the wallet name, such as the one given to you when you launched ygrunner.py.

Sure but it does say in the previous paragraph "it will print out  hex seed for a wallet you can use for tumble/sendpayment (just set the waletname to that hex value)". So it kind of does already tell you that.

> And replace 2Address with bcrt1qk48xnvsd9yma2yqgdxhfpcgny7fsz303z0hfv9?

Not sure why? Currently the software is doing all p2sh by default.

Reply:
> Sure but it does say in the previous paragraph "it will print out hex seed for a wallet you can use for tumble/sendpayment (just set the waletname to that hex value)". So it kind of does already tell you that.

Right, I was thinking this is more to make it clearer that's all.

> Not sure why? Currently the software is doing all p2sh by default.

O, I mean my aim was to make it clear that you are supposed to put a bitcoin address here that's all.

Reply:
OK, understood. I think I'll leave this as is, it's for developers after all (I don't think '2Address' will be unclear for them, and anyway they should know the basic syntax of sendpayment). We can always add more later.